compiler/src/iree/compiler/InputConversion/MHLO/MHLOToMHLOPreprocessing.cpp

```diff
@@ -419,87 +419,214 @@ class TransposeReshapeGenericDotGeneral
+
+    // Check indices vector has an implicit dim.
+    if (indexVectorDim != indicesTy.getRank()) {
+      return rewriter.notifyMatchFailure(op, "no implicit index dim");
+    }
+
+    // Materialize the implicit indices dim.
+    SmallVector<ReassociationExprs, 4> reassociationMap;
+    reassociationMap.resize(indicesTy.getRank());
+    SmallVector<int64_t> newShape;
+    for (int i = 0, s = indicesTy.getRank(); i < s; i++) {
+      reassociationMap[i].push_back(rewriter.getAffineDimExpr(i));
+      newShape.push_back(indicesTy.getDimSize(i));
+    }
+    if (!reassociationMap.empty()) {
+      reassociationMap.back().push_back(
+          rewriter.getAffineDimExpr(indicesTy.getRank()));
+    }
+    newShape.push_back(1);
+    indicesTy = RankedTensorType::get(newShape, indicesTy.getElementType());
+    indices = rewriter.create<tensor::ExpandShapeOp>(op.getLoc(), indicesTy,
+                                                     indices, reassociationMap);
+
+    auto newScatter = rewriter.create<mhlo::ScatterOp>(
+        op.getLoc(), op.getResultTypes(), op.operands(), indices, op.updates(),
+        dimNumbers, op.indices_are_sorted(), op.unique_indices());
+    Region &region = newScatter.update_computation();
+    rewriter.cloneRegionBefore(op.update_computation(), region, region.end());
+    rewriter.replaceOp(op, newScatter.getResults());
+    return success();
+  }
+};
+
+struct ScatterOpImplicitBatch : public OpRewritePattern<mhlo::ScatterOp> {
+  using OpRewritePattern<mhlo::ScatterOp>::OpRewritePattern;
+
+  static Value addUnitBatchDim(Location loc, Value value,
+                               PatternRewriter &rewriter) {
+    ShapedType valueTy = value.getType().cast<ShapedType>();
+    if (!valueTy.hasRank()) return nullptr;
+
+    // Materialize the implicit indices dim.
+    SmallVector<ReassociationExprs, 4> reassociationMap(valueTy.getRank());
+    if (!reassociationMap.empty()) {
+      reassociationMap.front().push_back(rewriter.getAffineDimExpr(0));
     }
 
+    SmallVector<int64_t> newShape = {1};
+    for (int i = 0, s = valueTy.getRank(); i < s; i++) {
+      reassociationMap[i].push_back(rewriter.getAffineDimExpr(i + 1));
+      newShape.push_back(valueTy.getDimSize(i));
+    }
+
+    valueTy = RankedTensorType::get(newShape, valueTy.getElementType());
+    return rewriter.create<tensor::ExpandShapeOp>(loc, valueTy, value,
+                                                  reassociationMap);
+  }
+
+  LogicalResult matchAndRewrite(mhlo::ScatterOp op,
+                                PatternRewriter &rewriter) const final {
     auto dimNumbers = op.scatter_dimension_numbers();
+    auto indexVectorDim = dimNumbers.getIndexVectorDim();
+    auto indices = op.scatter_indices();
+    auto indicesTy = indices.getType().cast<ShapedType>();
 
-    // We only have one dim for shape so this should be 0.
-    if (dimNumbers.getIndexVectorDim() != 0) return failure();
+    // Check whether indices
```

Comment:
I'm guessing this was an oversight?

Reply:
Just an incomplete comment. Probably was interrupted by something mid composition. Updated to new comment.